File: backend/routes/alerts.py
import uuid
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from core.database import get_db
from core.dependencies import get_current_user
from models.user import User
from schemas.alert import AlertOut
from services import alert_service

router = APIRouter(prefix="/alerts", tags=["alerts"])


@router.get("", response_model=list[AlertOut])
def list_alerts(
    unread_only: bool = Query(False),
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return alert_service.list_alerts(db, current_user, unread_only)


@router.patch("/{alert_id}/read", response_model=AlertOut)
def mark_read(
    alert_id: uuid.UUID,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    return alert_service.mark_alert_read(db, current_user, alert_id)

File: frontend/src/api/alerts.js
import api from "./client";

export async function listAlerts(unreadOnly = false) {
  const { data } = await api.get("/alerts", { params: { unread_only: unreadOnly } });
  return data;
}

export async function markAlertRead(id) {
  const { data } = await api.patch(`/alerts/${id}/read`);
  return data;
}
